File: Codes/CommonString/CommonString.hs
import Data.List

getTokens xs n = [take n (drop i xs) | i <- [0..(length xs - n)]]

findCommonString s1 s2 = if length result > 0 then head result else ""
    where
        (shortStr, longStr) = if length s1 < length s2 then (s1, s2) else (s2, s1)
        shortLength = length shortStr
        result = [token | n <- reverse [1..shortLength], token <- getTokens shortStr n, isInfixOf token longStr]

-- let's find common string.     
main = return $ findCommonString "photography" "autograph"

File: Codes/CommonString/CommonString.py
def findCommonString(s, lowerLengthString):
    if len(lowerLengthString) > len(s):
        s, lowerLengthString = lowerLengthString, s

    n = len(lowerLengthString)
    for i in range(n):
        for j in range(i+1):
            token = lowerLengthString[j:n-i+j]
            if token in s:
                print(len(token))
                print(token)
                return
    else:
        print("Not found.")

# let's find common string.
findCommonString("photography", "autograph")
